repos/system_upgrade/common/actors/scaninstalledtargetkernelversion/libraries/scankernel.py

```diff
@@ -1,53 +1,80 @@
+    if not src_kernel_info:
+        return  # Will not happen, other actors would inhibit the upgrade
+
+    target_ver = get_target_major_version()
+    target_kernel_pkg_name = kernel_lib.get_kernel_pkg_name(target_ver, src_kernel_info.type)
+    target_kernel_nevra = get_target_kernel_package_nevra(target_kernel_pkg_name)
+
+    if src_kernel_info.type != kernel_lib.KernelType.ORDINARY and not target_kernel_nevra:
+        api.current_logger().warning('The kernel-rt-core rpm from the target RHEL has not been detected. Switching '
+                                     'to non-preemptive kernel.')
+        target_kernel_pkg_name = kernel_lib.get_kernel_pkg_name(target_ver, kernel_lib.KernelType.ORDINARY)
+        target_kernel_nevra = get_target_kernel_package_nevra(target_kernel_pkg_name)
+
+    if target_kernel_nevra:
+        boot_files = get_boot_files_provided_by_kernel_pkg(target_kernel_nevra)
+        installed_kernel_info = InstalledTargetKernelInfo(pkg_nevra=target_kernel_nevra,
+                                                          kernel_img_path=boot_files.vmlinuz_path,
+                                                          initramfs_path=boot_files.initramfs_path)
+        api.produce(installed_kernel_info)
 
-    if is_rhel_realtime():
-        version = _get_kernel_version('kernel-rt-core')
-        if version:
-            api.produce(InstalledTargetKernelVersion(version=version))
-            return
-        else:
-            api.current_logger().warning(
-                'The kernel-rt-core rpm from the target RHEL has not been detected. '
-                'Switching to non-preemptive kernel.'
-            )
-            # TODO: create report with instructions to install kernel-rt manually
-            # - attach link to article if any
-            # - this possibly happens just in case the repository with kernel-rt
-            # # is not enabled during the upgrade.
-
-    # standard (non-preemptive) kernel
-    version = _get_kernel_version('kernel-core')
-    if version:
+        # Backwards compatibility
+        # Expects that the kernel nevra has the following format: <pkg_name>-<version>-<release>.<arch>
+        version = '-'.join(target_kernel_nevra.split('-')[-2:])  # (-2)-th is <version>; take <version>-<release>...
         api.produce(InstalledTargetKernelVersion(version=version))
     else:
-        # This is very unexpected situation. At least one kernel has to be
-        # installed always. Some actors consuming the InstalledTargetKernelVersion
-        # will crash without the created message. I am keeping kind of original
-        # behaviour in this case, but at least the let me log the error msg
-        #
-        api.current_logger().error('Cannot detect any kernel RPM')
-        # StopActorExecutionError('Cannot detect any target RHEL kernel RPM.')
+        raise StopActorExecutionError('Cannot detect any target RHEL kernel RPM.')  # Unexpected.
```

Comment:
@MichalHe: note to self: this should be commented out, because at this stage it is too late to throw anything - stopping would do only harm (better have 80% and continue than fail completely)